py4DSTEM/process/diffraction/crystal_ACOM.py

```diff
@@ -1430,6 +1433,132 @@ def match_single_pattern(
 
 
 
+
+def calculate_strain(
+    self,
+    bragg_peaks_array: PointListArray,
+    orientation_map: OrientationMap,
+    corr_kernel_size = None,
+    sigma_excitation_error = 0.02,
+    min_num_peaks = 5,
+    rotation_range = None,
+    progress_bar = True,
+    ):
+    '''
+    This function takes in both a PointListArray containing Bragg peaks, and a 
+    corresponding OrientationMap, and uses least squares to compute the 
+    deformation tensor which transforms the simulated diffraction pattern
+    into the experimental pattern, for all probe positons.
+
+    TODO: add robust fitting?
+
+    Args: 
+        bragg_peaks_array (PointListArray):
+        orientation_map (OrientationMap):
+        corr_kernel_size (float):
+
+    Returns:
+        strain_map (RealSlice):  strain tensor
+
+    '''
+
+    # Initialize empty strain maps
+    strain_map = RealSlice(
+        data=np.zeros((
+            bragg_peaks_array.shape[0],
+            bragg_peaks_array.shape[1],
+            5)),
+        slicelabels=('e_xx','e_yy','e_xy','theta','mask'),
+        name='strain_map')
+    strain_map.slices['mask'] = 1
+
+    # init values
+    if corr_kernel_size is None:
+        corr_kernel_size = self.orientation_kernel_size
+    radius_max_2 = corr_kernel_size**2
+
+    # Loop over all probe positions
+    for rx, ry in tqdmnd(
+        # 10,10,
+        # range(45,55),
+        # range(220,230),
+        # range(0,bragg_peaks_array.shape[0],1),
+        # range(120,130),
+        *bragg_peaks_array.shape,
+        desc="Calculating strains",
+        unit=" PointList",
+        disable=not progress_bar,
+        ):
+        # Get bragg peaks from experiment and reference
+        p = bragg_peaks_array.get_pointlist(rx,ry)
+
+        if p.data.shape[0] >= min_num_peaks:
+            p_ref = self.generate_diffraction_pattern(
+                orientation_map.get_orientation(rx,ry),
```

Comment:
I think we should also let the user specify the other relevant arguments to `generate_diffraction_pattern`, i.e. `tol_excitation_error_mult` and `k_max`

Reply:
Done and done!  All default options copied over, with exact same argument names.